dialog/fileitem.go

```diff
@@ -80,13 +71,13 @@ func (f *fileDialog) newFileItem(location fyne.URI, dir bool) *fileDialogItem {
 	item := &fileDialogItem{
 		picker:   f,
 		location: location,
+		name:     location.Name(),
 		dir:      dir,
 	}
 
-	if dir {
-		item.name = location.Name()
-	} else {
-		item.name = fileName(location)
+	if f.view == gridView {
+		ext := filepath.Ext(item.name[1:])
```

Comment:
I think this might crash for a file named "H"?

Reply:
It seems like it actually doesn't. We were doing the same thing in the code that I removed.
![image](https://user-images.githubusercontent.com/25466657/171880074-155aee10-2974-43ee-aac4-909601544f14.png)

Reply:
Oops! thanks